crates/bevy_render/src/texture/image.rs

```diff
@@ -251,6 +251,25 @@ impl Image {
         value
     }
 
+    /// Create a new zero-filled image with a given size, which can be rendered to.
+    /// This is primarily for use as a render target for a [`Camera`](`crate::camera::Camera`).
+    /// See [`RenderTarget::Image`](`crate::camera::RenderTarget::Image`).
+    pub fn new_target_texture(width: u32, height: u32) -> Self {
+        let mut image = Self::default();
+
+        image.resize(Extent3d {
+            width,
+            height,
+            ..Default::default()
+        });
```

Comment:
Technically `resize` *extends* the data with zeroes (or truncates it).
And actually for the `Default` impl `data = vec![255; 4]`, so you actually end up with `[255, 255, 255, 255, 0, 0, 0, 0, ...]` which is probably not really what's intended, but that's how the example was originally constructed.
Maybe the `Default` impl should be zeroes `data = vec![0; 4] but I don't know if there's any implications with that.

Alternatively for this method, we could build the data directly, rather than calling resize at all. Presumably the compiler would optimise out the first assignment to data.
```rust
    pub fn new_target_texture(width: u32, height: u32) -> Self {
        let mut image = Self::default();

        let size = Extent3d {
            width,
            height,
            ..Default::default()
        };

        image.data = vec![0; size.volume() * image.texture_descriptor.format.pixel_size()];
        image.texture_descriptor.size = size;
        
        image.texture_descriptor.usage = wgpu::TextureUsages::TEXTURE_BINDING
            | wgpu::TextureUsages::COPY_DST
            | wgpu::TextureUsages::RENDER_ATTACHMENT;

        image
    }
```

I'm not sure about additional methods, but something like this sort of makes sense:
```rust
    pub fn resize(&mut self, size: Extent3d) {
        self.resize_with(&[0]);
    }
    pub fn resize_with(&mut self, size: Extent3d, pixel: &[u8]) {
        todo!();
    }
    pub fn fill_with(&mut self, pixel: &[u8]) {
        for current_pixel in self.data.chunks_exact_mut(pixel.len()) {
            current_pixel.copy_from_slice(pixel);
        }
    }
    pub fn fill_with_zeroes(&mut self) {
        self.fill_with(&[0]);
    }
```

Unless we're intending to use these methods to implement this method, could this be a separate PR?

Reply:
Yes, setting the data directly I guess would work too. 
Not sure about the current default, it is a bit odd indeed. Maybe let's hear what the others say.

Regarding extra methods, it can definitely be deferred to another PR and be discussed there.